src/dwarf_eh.h

```diff
@@ -237,9 +249,15 @@ static uint64_t read_value(char encoding, dw_eh_ptr_t *data)
 		READ(DW_EH_PE_sdata4, int32_t)
 		READ(DW_EH_PE_sdata8, int64_t)
 #ifdef __CHERI_PURE_CAPABILITY__
-		READ(DW_EH_PE_absptr, int64_t)
+        case DW_EH_PE_absptr:
+                {
+                        intptr_t t;
```

Comment:
Since the function returns uintptr_t I guess this should also be uintptr_t to avoid some -Wpedantic/-Wconversion warnings

Reply:
Huh, yeah, that was me not being fully consistent with s/intrptr_t/uintptr_t/ in this function.  I'll also fix the assert_pointer_in_bounds() helpers to use uintptr_t as those are the only other things now using 'intptr_t' in this file.